contracts/src/v0.8/dev/UpkeepTranscoder.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./interfaces/UpkeepTranscoderInterface.sol";
+import "../ConfirmedOwner.sol";
+
+/**
+ * @notice Transcoder for converting upkeep data from one keeper
+ * registry version to another
+ */
+contract UpkeepTranscoder is UpkeepTranscoderInterface, ConfirmedOwner {
+  constructor() ConfirmedOwner(msg.sender) {}
+
+  /**
+   * @notice transcodeUpkeeps transforms upkeep data from the format expected by
+   * one registry to the format expected by another. It future-proofs migrations
+   * by allowing keepers team to customize migration paths and set sensible defaults
+   * when new fields are added
+   * @param fromRegistry registry the upkeep is migrating from
+   * @param toRegistry registry the upkeep is migrating to
+   * @param encodedUpkeeps encoded upkeep data
+   * @dev this contract & function are simple now, but should evolve as new registries
+   * and migration paths are added
```

Comment:
(y) I like the idea

Reply:
Agreed. Cool forward thinking plan. I think it may be less effort long term to pass a `fromType` and `toType` or some kind of enumerables instead of having to register each specific address pair. Then Registries can just declare their importing type format and exporting registries can look it up. It wouldn't even have to be looked up, if we switch to storing some kind of struct with info for each registry the registry owner could set it when they permission another registry for migration.

Reply:
I think this design supports that. ex if each registry has a `type()` getter, then the transcoder can read that from the contract and adjust transpose accordingly. My goal was to remove the need for each registry to know about the "type" of every other registry, and just let the transcoder handle that. I think "type" also might not be enough for example if migrating b/t registries with different max/min values. They may have the same data "type" but we need the intermediary to adjust the values sensibly.

Reply:
If there's 4 registries with 2 different types, we have to add 24 records in the Transcoder for how to migrate between them. If the transcoder only knows type, we just have 3 paths. I think address may be too specific, enough to be burdensome.

Reply:
I think we're saying the same thing here. I'm just suggesting you have an `address => type` map.